nvflare/app_common/aggregators/dxo_aggregator.py

```diff
@@ -0,0 +1,152 @@
+
+        processed_algorithm = dxo.get_meta_prop(MetaKey.PROCESSED_ALGORITHM)
+        if processed_algorithm is not None:
+            self.log_error(fl_ctx, f"unable to accept DXO processed by {processed_algorithm}")
+            return False
+
+        current_round = fl_ctx.get_prop(AppConstants.CURRENT_ROUND)
+        self.log_debug(fl_ctx, f"current_round: {current_round}")
+
+        data = dxo.data
+        if data is None:
+            self.log_error(fl_ctx, "no data to aggregate")
+            return False
+
+        n_iter = dxo.get_meta_prop(MetaKey.NUM_STEPS_CURRENT_ROUND)
+        if contribution_round != current_round:
+            self.log_warning(
+                fl_ctx,
+                f"discarding DXO from {contributor_name} at round: "
+                f"{contribution_round}. Current round is: {current_round}",
+            )
+            return False
+
+        for item in self.aggregation_helper.get_history():
+            if contributor_name == item["contributor_name"]:
+                prev_round = item["round"]
+                self.log_warning(
+                    fl_ctx,
+                    f"discarding DXO from {contributor_name} at round: "
+                    f"{contribution_round} as {prev_round} accepted already",
+                )
+                return False
+
+        if n_iter is None:
+            if self.warning_count.get(contributor_name, 0) <= self.warning_limit:
+                self.log_warning(
+                    fl_ctx,
+                    f"NUM_STEPS_CURRENT_ROUND missing in meta of DXO"
+                    f" from {contributor_name} and set to default value, 1.0. "
+                    f" This kind of message will show {self.warning_limit} times at most.",
+                )
+                if contributor_name in self.warning_count:
+                    self.warning_count[contributor_name] = self.warning_count[contributor_name] + 1
+                else:
+                    self.warning_count[contributor_name] = 0
+            n_iter = 1.0
+        float_n_iter = float(n_iter)
+        aggregation_weight = self.aggregation_weights.get(contributor_name)
+        if aggregation_weight is None:
+            if self.warning_count.get(contributor_name, 0) <= self.warning_limit:
+                self.log_warning(
+                    fl_ctx,
+                    f"Aggregation_weight missing for {contributor_name} and set to default value, 1.0"
+                    f" This kind of message will show {self.warning_limit} times at most.",
+                )
+                if contributor_name in self.warning_count:
+                    self.warning_count[contributor_name] = self.warning_count[contributor_name] + 1
+                else:
+                    self.warning_count[contributor_name] = 0
+            aggregation_weight = 1.0
+
+        # aggregate
+        self.aggregation_helper.add(data, aggregation_weight * float_n_iter, contributor_name, contribution_round)
```

Comment:
we pass "aggregation_weight * float_n_iter" here

Reply:
This logic is different from previous implementation, so we need to update the test cases.

Reply:
Added a new test script for the new intime aggregation logic.